base/shell/cmd/misc.c

```diff
@@ -184,16 +184,18 @@ BOOL add_entry (LPINT ac, LPTSTR **arg, LPCTSTR entry)
     LPTSTR *oldarg;
 
     q = cmd_alloc ((_tcslen(entry) + 1) * sizeof (TCHAR));
-    if (NULL == q)
+    if (!q)
     {
+        WARN("DEBUG: Cannot allocate memory for q!\n");
         return FALSE;
     }
 
     _tcscpy (q, entry);
     oldarg = *arg;
     *arg = cmd_realloc (oldarg, (*ac + 2) * sizeof (LPTSTR));
-    if (NULL == *arg)
+    if (!*arg)
     {
+        WARN("DEBUG: Cannot reallocate memory for arg!\n");
```

Comment:
oldarg is still valid here?

Reply:
cmd_realloc should iirc. behave as the regular realloc: if it fails, the old pointer is still valid.